subprojects/snapshots/src/main/java/org/gradle/internal/fingerprint/FingerprintingStrategy.java

```diff
@@ -47,6 +47,16 @@
      */
     String getIdentifier();
 
+    /**
+     * UsedByScansPlugin
+     */
+    DirectorySensitivity getDirectorySensitivity();
+
+    /**
+     * UsedByScansPlugin
```

Comment:
I think this is more harmful than helpful. The relevant usages of this are in the codebase, and the scan plugin doesn't actually use this in any meaningful way. Said another way, anything about these methods can change, as long as the operation interface doesn't change.